velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -426,6 +433,69 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, reclaimFromOrderBy) {
   }
 }
 
+DEBUG_ONLY_TEST_F(
+    SharedArbitrationTest,
+    filterProjectInitTriggeredArbitration) {
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::memory::{}::FakeMemoryOperator::FakeMemoryOperator",
+      std::function<void(std::pair<MemoryPool*, memory::Allocation*>*)>(
+          [&](std::pair<MemoryPool*, memory::Allocation*>* allocationHook) {
+            allocationHook->first->allocateNonContiguous(
```

Comment:
plan buffer allocation should be enough

Reply:
plan buffer allocation (void*) cannot be destructed automatically. We need to add another field in the FakeMemoryOperator to store that buffer then clear it upon destruction manually. I'm taking advantage of Allocation's auto cleanup mechanism here. Plus there is already an allocations_ field inside (I did not notice previously but I'll change to use that, plus ctor cb), no extra work.